cabal-install/Distribution/Client/List.hs

```diff
@@ -138,27 +235,39 @@ list :: Verbosity
      -> ListFlags
      -> [String]
      -> IO ()
-list verbosity packageDBs repos comp conf listFlags pats = do
-    matches <- getPkgList verbosity packageDBs repos comp conf listFlags pats
-
-    if simpleOutput
-      then putStr $ unlines
-             [ display (pkgName pkg) ++ " " ++ display version
-             | pkg <- matches
-             , version <- if onlyInstalled
-                            then              installedVersions pkg
-                            else nub . sort $ installedVersions pkg
-                                           ++ sourceVersions    pkg ]
-             -- Note: this only works because for 'list', one cannot currently
-             -- specify any version constraints, so listing all installed
-             -- and source ones works.
-      else
-        if null matches
-            then notice verbosity "No matches found."
-            else putStr $ unlines (map showPackageSummaryInfo matches)
+list verbosity packageDBs repoCtxt comp conf listFlags pats = do
+  unless (simpleOutput || null fields) $
```

Comment:
So just `--field=foo --simple-output` is allowed, even though it has no effect? From the error message it looks like the test should be `when (not.null fields && (not simpleOutput || not onlyDependencies)) $ die "..."`.